src/stream/src/executor/merge.rs

```diff
@@ -200,12 +217,28 @@ impl Executor for MergeExecutor {
     }
 }
 
+#[derive(Default)]
+struct StagedWatermarks {
+    in_heap: bool,
+    staged: VecDeque<Watermark>,
+}
+
+struct UntransferredWatermarks {
```

Comment:
How about renaming it as `UnemittedWatermarks` or `BufferedWatermarks`, which is more common in our system. :)

Reply:
> How about renaming it as `UnemittedWatermarks` or `BufferedWatermarks`, which is more common in our system. :)

resolved